tests/containers/podman_netavark.pm

```diff
@@ -0,0 +1,147 @@
+        return 1;
+    }
+
+    switch_to_netavark;
+    $podman->cleanup_system_host();
+
+    ## TEST1
+    record_info('TEST1', 'set static IP, and MAC addresses on a per-network basis');
+    my $testnet = 'test_simple';
+    my $test_ctr = get_required_var('REGISTRY') . '/fedora/apache';
+
+    assert_script_run("podman network create --gateway 10.90.0.1 --subnet 10.90.0.0/16 $testnet");
+    assert_script_run("podman run --network $testnet:ip=10.90.0.8,mac=76:22:33:44:55:66 -d --name test_apache $test_ctr");
+    assert_script_run("podman container inspect test_apache --format {{.NetworkSettings.Networks.$testnet.IPAddress}}");
+    validate_script_output("curl -I -s 10.90.0.8:80", sub { /HTTP.* 200 OK/ });
+    assert_script_run("grep test_apache /run/containers/networks/aardvark-dns/$testnet");
+    remove_subtest_setup;
+
+    ## TEST2
+    record_info('TEST2', 'set static IPs for containers and check their networking');
+    $testnet = 'primary';
+    my $testnet2 = 'secondary';
+    my $test_ctr_sec = 'registry.opensuse.org/bci/bci-busybox';
+
+    assert_script_run("podman network create --gateway 10.90.0.1 --subnet 10.90.0.0/16 $testnet");
+    assert_script_run("podman network create --gateway 10.64.0.1 --subnet 10.64.0.0/16 $testnet2");
+    assert_script_run("podman run --network $testnet:ip=10.90.0.8,mac=76:22:33:44:55:66 -d --name test_apache $test_ctr");
+    assert_script_run("podman run --network $testnet2:ip=10.64.0.8,mac=92:aa:33:44:55:66 --network $testnet:ip=10.90.0.64,mac=92:bb:cc:44:55:66 -dt --name test_second $test_ctr_sec");
+    # second container should have 2 interfaces
+    my $net1_reg = qr@ether\s+92:aa:33:44:55:66.*\s+inet\s+10\.64\.0\.8\/16@;
+    my $net2_reg = qr@ether\s+92:bb:cc:44:55:66.*\s+inet\s+10\.90\.0\.64\/16@;
+    validate_script_output("podman exec -t test_second /bin/sh -c 'ip addr show eth0'", sub { /$net1_reg|$net2_reg/m });
+    validate_script_output("podman exec -t test_second /bin/sh -c 'ip addr show eth1'", sub { /$net1_reg|$net2_reg/m });
+    validate_script_output("podman exec -t test_second /bin/sh -c 'nslookup test_apache'", sub { /Name:\s+test_apache\s+Address.*10.90.0.8/m });
+    validate_script_output("podman exec -t test_second /bin/sh -c 'wget -S 10.90.0.8:80'", sub { /HTTP.* 200 OK/ });
+    assert_script_run("grep test_apache /run/containers/networks/aardvark-dns/$testnet");
+    assert_script_run("grep test_second /run/containers/networks/aardvark-dns/$testnet");
+    assert_script_run("grep test_second /run/containers/networks/aardvark-dns/$testnet2");
+    remove_subtest_setup;
+
+    ## TEST3
+    record_info('TEST3', 'create a dual stack network');
+    $testnet = 'test_dual_stack';
+    assert_script_run("podman network create --ipv6 --gateway fd00::1:8:1 --subnet fd00::1:8:0/112 --gateway 10.90.0.1 --subnet 10.90.0.0/16 $testnet");
+    assert_script_run("podman network inspect $testnet");
```

Comment:
what is this line testing? if it's testing the successful creation of the network in the previous step, shouldn't `assert_script_run` take care of it? I'm not sure hence asking

Reply:
Good catch, it was my debug output while writing the test.